tests/model_fields/test_onetoonefield.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import unicode_literals
+
+from django.db import models
+from django.test import SimpleTestCase
+from django.test.utils import isolate_apps
+
+
+class OneToOneFieldTests(SimpleTestCase):
+    @isolate_apps('model_fields')
+    def test_abstract_app_relative_reference(self):
+        class A(models.Model):
+            pass
+
+        class B(A):
+            a = models.OneToOneField('A', on_delete=models.CASCADE)
+
+            class Meta:
+                abstract = True
+
+        class C(B):
+            pass
+
+        self.assertIs(C._meta.parents[A], C._meta.get_field('a'))
```

Comment:
This test method will be added to the appropriate class when backporting to the `stable/1.9.x` branch.